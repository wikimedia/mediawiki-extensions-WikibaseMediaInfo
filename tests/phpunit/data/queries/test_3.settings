{
	"title": "No entities found, lang fallbacks, stemming",
	"term": "yet another search term",
	"entitiesForTerm": {
		"yet another search term": []
	},
	"userLanguage": "de",
	"fallbackLangs": [ "fr", "es" ],
	"stemmingSettings": {
		"ga": {
			"query": true
		},
		"fr": {
			"query": true
		},
		"de": {
			"query": true
		}
	},
	"settings": {
		"boost": {
			"statement": {"P999": 20},
			"descriptions.$language": 7.5,
			"descriptions.$language.plain": 2.5,
			"category": 0.525,
			"category.plain": 0.175,
			"title": 0.45,
			"title.plain": 0.15,
			"redirect.title": 0.375,
			"redirect.title.plain": 0.125,
			"text": 0.3,
			"text.plain": 0.1,
			"auxiliary_text": 0.225,
			"auxiliary_text.plain": 0.075,
			"file_text": 0.15,
			"file_text.plain": 0.05
		}
	}
}
