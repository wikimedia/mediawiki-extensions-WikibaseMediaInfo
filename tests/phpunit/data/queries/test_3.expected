{
    "bool": {
        "filter": [
            {
                "bool": {
                    "should": [
                        {
                            "multi_match": {
                                "fields": [
                                    "all",
                                    "all.plain",
                                    "descriptions.de",
                                    "descriptions.fr"
                                ],
                                "query": "yet another search term",
                                "operator": "and"
                            }
                        }
                    ]
                }
            }
        ],
        "should": [
            [
                {
                    "function_score": {
                        "query": {
                            "bool": {
                                "should": [
                                    {
                                        "function_score": {
                                            "query": {
                                                "bool": {
                                                    "should": [
                                                        {
                                                            "bool": {
                                                                "should": [
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "yet another search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "descriptions.de^3",
                                                                                "descriptions.de.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "yet another search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "descriptions.fr^3",
                                                                                "descriptions.fr.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "yet another search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "descriptions.es.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "yet another search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "title^3",
                                                                                "title.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "yet another search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "category^3",
                                                                                "category.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "yet another search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "redirect.title^3",
                                                                                "redirect.title.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "yet another search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "suggest"
                                                                            ]
                                                                        }
                                                                    }
                                                                ]
                                                            }
                                                        },
                                                        {
                                                            "bool": {
                                                                "should": [
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "yet another search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "heading^3",
                                                                                "heading.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "yet another search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "auxiliary_text^3",
                                                                                "auxiliary_text.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "yet another search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "text^3",
                                                                                "text.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "yet another search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "file_text^3",
                                                                                "file_text.plain^1"
                                                                            ]
                                                                        }
                                                                    }
                                                                ]
                                                            }
                                                        }
                                                    ]
                                                }
                                            },
                                            "functions": [
                                                {
                                                    "script_score": {
                                                        "script": {
                                                            "source": "max(0, ln(_score))",
                                                            "lang": "expression"
                                                        }
                                                    }
                                                }
                                            ]
                                        }
                                    },
                                    {
                                        "function_score": {
                                            "query": {
                                                "match_explorer": {
                                                    "type": "unique_terms_count",
                                                    "query": {
                                                        "match": {
                                                            "text": {
                                                                "query": "yet another search term"
                                                            }
                                                        }
                                                    }
                                                }
                                            },
                                            "functions": [
                                                {
                                                    "script_score": {
                                                        "script": {
                                                            "source": "ln(1 \/ max(1.25, _score))",
                                                            "lang": "expression"
                                                        }
                                                    }
                                                }
                                            ]
                                        }
                                    }
                                ]
                            }
                        },
                        "functions": [
                            {
                                "script_score": {
                                    "script": {
                                        "source": "exp(_score)",
                                        "lang": "expression"
                                    }
                                }
                            }
                        ]
                    }
                },
                {
                    "bool": {
                        "must": [
                            {
                                "multi_match": {
                                    "fields": [
                                        "all_near_match^2",
                                        "all_near_match.asciifolding^1.5"
                                    ],
                                    "query": "yet another search term",
                                    "operator": "and"
                                }
                            }
                        ],
                        "filter": [
                            {
                                "terms": {
                                    "namespace": [
                                        0,
                                        14
                                    ]
                                }
                            }
                        ],
                        "boost": 1
                    }
                }
            ]
        ]
    }
}